src/db/utils/address.cpp

```diff
@@ -1,39 +1,36 @@
-//
-// Created by zanna on 05/10/18.
-//
-
 #include "address.h"
 
-const char* const db::utils::Address::PORT_SEPARATOR = ":";
-const char* const db::utils::Address::PATH_SEPARATOR = "/";
-const std::string db::utils::Address::HTTP_PREFIX = "http://";
+const char *const db::utils::Address::PORT_SEPARATOR = ":";
+const char *const db::utils::Address::PATH_SEPARATOR = "/";
+const std::string db::utils::Address::HTTP_PREFIX("http://");
 
-db::utils::Address::Address(const std::string& new_address, uint16_t new_port)
+db::utils::Address::Address(const std::string &new_address, uint16_t new_port)
         : address(new_address), port(new_port), has_http_prefix(false) {
     has_http_prefix = prefix_check(new_address);
 }
 
-db::utils::Address::Address(const std::string& complete_address) {
+db::utils::Address::Address(const std::string &complete_address) {
     has_http_prefix = prefix_check(complete_address);
     if (!has_http_prefix) {
         address_parse(complete_address);
     } else {
         size_t found = complete_address.find_first_of(PATH_SEPARATOR);
-        std::string protocol = complete_address.substr(0,found);
-        std::string url_new = complete_address.substr(found+3);
+        std::string protocol = complete_address.substr(0, found);
+        std::string url_new = complete_address.substr(found + 3);
         address_parse(url_new);
     }
 }
 
-void db::utils::Address::address_parse(const std::string& url) {
+void db::utils::Address::address_parse(const std::string &url) {
     size_t found = url.find_first_of(PORT_SEPARATOR);
-    address = url.substr(0,found);
+    address = url.substr(0, found);
     size_t found1 = url.find_first_of(PATH_SEPARATOR);
-    port = std::stoul(url.substr(found+1,found1-found-1));
+    port = (uint16_t) (std::stoul(url.substr(found + 1, found1 - found - 1)));
```

Comment:
Should use `static_const` but whatever, our code is full of C casts (so it's really not necessary)...
```suggestion
    port = static_cast<uint16_t>(std::stoul(url.substr(found + 1, found1 - found - 1)));
```

Reply:
done